public/app/features/connections/pages/AddNewConnectionPage.tsx

```diff
@@ -1,12 +1,45 @@
+import { useState } from 'react';
+
+import { PluginType } from '@grafana/data';
+import { Button } from '@grafana/ui';
 import { Page } from 'app/core/components/Page/Page';
+import { Trans } from 'app/core/internationalization';
+import UpdateAllModal from 'app/features/plugins/admin/components/UpdateAllModal';
+import { useGetUpdatable } from 'app/features/plugins/admin/state/hooks';
 
 import { AddNewConnection } from '../tabs/ConnectData';
 
 export function AddNewConnectionPage() {
+  const { isLoading: areUpdatesLoading, updatablePlugins } = useGetUpdatable();
+  const updatableDSPlugins = updatablePlugins.filter((plugin) => plugin.type === PluginType.datasource);
+  const [showUpdateModal, setShowUpdateModal] = useState(false);
+  const disableUpdateAllButton = updatableDSPlugins.length <= 0 || areUpdatesLoading;
+
+  const onUpdateAll = () => {
+    setShowUpdateModal(true);
+  };
+
+  const updateAll = (
+    <Button disabled={disableUpdateAllButton} onClick={onUpdateAll}>
+      {disableUpdateAllButton ? (
+        <Trans i18nKey="plugins.catalog.no-updates-available">No updates available</Trans>
+      ) : (
+        <Trans i18nKey="plugins.catalog.update-all.button" values={{ length: updatablePlugins.length }}>
+          Update all ({{ length }})
+        </Trans>
+      )}
+    </Button>
+  );
```

Comment:
Instead of defining this inline, can we move it to a separate file and expose it as it own component?

Reply:
Also, this code seems to be duplicated so then it makes even more sense to break it out into a single component and re-use it. If I don't read the code incorrectly.